src/github.com/matrix-org/dendrite/federationapi/routing/routing.go

```diff
@@ -85,6 +85,16 @@ func Setup(
 		},
 	))
 
+	v1fedmux.Handle("/exchange_third_party_invite/{roomID}", common.MakeFedAPI(
+		"exchange_third_party_invite", cfg.Matrix.ServerName, keys,
+		func(httpReq *http.Request, request *gomatrixserverlib.FederationRequest) util.JSONResponse {
+			vars := mux.Vars(httpReq)
+			return writers.ExchangeThirdPartyInvite(
+				httpReq, request, vars["roomID"], query, cfg, federation, producer,
+			)
+		},
+	))
```

Comment:
After reading #223 I think you want to add `.Methods("PUT", "OPTIONS")` here

Reply:
I was planning on doing a whole other PR this week to address this issue Dendrite-wide, that's why I didn't bother with it in this one.

Reply:
all right then